File: frontend/src/types/prompt.ts
import { Category, Tag } from './categoryTag';
import { User } from './auth';

export interface Prompt {
  id: number;
  title: string;
  description?: string;
  promptText: string;
  systemInstruction?: string;
  targetModel: string;
  
  // AI Tool & External Chat Location Properties
  aiTool: string;
  externalChatUrl?: string;
  externalChatId?: string;
  chatSummary?: string;

  isPublic: boolean;
  isFavorite: boolean;
  viewCount: number;
  likeCount: number;
  category: Category;
  tags: Tag[];
  author: User;
  extractedVariables: string[];
  createdAt: string;
  updatedAt?: string;
}

export interface PromptRequest {
  title: string;
  description?: string;
  promptText: string;
  systemInstruction?: string;
  targetModel: string;
  
  aiTool?: string;
  externalChatUrl?: string;
  externalChatId?: string;
  chatSummary?: string;

  categoryId: number;
  tagIds?: number[];
  isPublic?: boolean;
}

export interface PagedResponse<T> {
  content: T[];
  page: number;
  size: number;
  totalElements: number;
  totalPages: number;
  last: boolean;
}

File: backend/src/main/java/com/promptvault/entity/Prompt.java
package com.promptvault.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "prompts")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Prompt {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String title;

    @Column(length = 500)
    private String description;

    @Lob
    @Column(name = "prompt_text", nullable = false)
    private String promptText;

    @Lob
    @Column(name = "system_instruction")
    private String systemInstruction;

    @Column(name = "target_model", length = 50)
    @Builder.Default
    private String targetModel = "GPT-4";

    // AI Tool Integration & External Chat Location Indexing
    @Column(name = "ai_tool", length = 50)
    @Builder.Default
    private String aiTool = "ChatGPT";

    @Column(name = "external_chat_url", length = 500)
    private String externalChatUrl;

    @Column(name = "external_chat_id", length = 100)
    private String externalChatId;

    @Column(name = "chat_summary", length = 500)
    private String chatSummary;

    @Column(name = "is_public", nullable = false)
    @Builder.Default
    private Boolean isPublic = true;

    @Column(name = "is_favorite", nullable = false)
    @Builder.Default
    private Boolean isFavorite = false;

    @Column(name = "view_count", nullable = false)
    @Builder.Default
    private Integer viewCount = 0;

    @Column(name = "like_count", nullable = false)
    @Builder.Default
    private Integer likeCount = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "prompt_tags",
            joinColumns = @JoinColumn(name = "prompt_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id"))
    @Builder.Default
    private Set<Tag> tags = new HashSet<>();

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
